vertx-template-engines/vertx-web-templ-pebble/src/main/java/io/vertx/ext/web/templ/impl/PebbleVertxLoader.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2016 Original authors.
+ *
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *  The Eclipse Public License is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  The Apache License v2.0 is available at
+ *  http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.ext.web.templ.impl;
+
+import com.mitchellbosecke.pebble.error.LoaderException;
+import com.mitchellbosecke.pebble.loader.Loader;
+import io.vertx.core.Vertx;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.charset.Charset;
+
+/**
+ * A PebbleLoader based on Vertx
+ *
+ * @author Paulo Lopes
+ */
+class PebbleVertxLoader implements Loader<String> {
+
+  private final Vertx vertx;
+
+  private String charset = Charset.defaultCharset().toString();
+
+  public PebbleVertxLoader(Vertx vertx) {
```

Comment:
This is actually a big change. Before the template were loaded from the classpath so could be embedded in a fat jar or in another jar (present on the classpath). I'm not sure it will work with this loader.